frontend/src/lib/WebsocketConnection.tsx

```diff
@@ -346,7 +352,23 @@ export class WebsocketConnection {
     }
 
     logMessage(LOG, "creating WebSocket")
-    this.websocket = new WebSocket(uri)
+
+    // NOTE: We repurpose the Sec-WebSocket-Protocol header here in a slightly
+    // unfortunate (but necessary) way. The browser WebSocket API doesn't
+    // allow us to set arbitrary HTTP headers, and this header is the only one
+    // where we have the ability to set it to arbitrary values, so we use it
+    // to pass an auth token from client to server as the *second* value in the
+    // list.
+    //
+    // The reason why the auth token is set as the second value is that, when
```

Comment:
To clarify: Adding a list of protocols is generally supported capabilities of WebSocket? I assume the server is expected to choose one of the protocols in the list and return this as value?

Reply:
Yep. It's the only arbitrarily-settable header value that we can control from javascript. I added a comment clarifying that the second argument to the `WebSocket` constructor is what populates the header value.